src/main/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImpl.java

```diff
@@ -26,111 +27,130 @@
+    return accountMapper.accountToAccountDto(savedAccount);
+  }
 
-    @Override
-    public AccountDto getAccountByEmail(String email) {
-        Account account = accountRepository.findOneByEmail(email)
-                .orElseThrow(() -> new AccountNotFoundException(AccountConstant.NOT_FOUND));
+  @Override
+  public AccountDto getAccountByEmail(String email) {
+    Account account = accountRepository.findOneByEmail(email)
+        .orElseThrow(() -> new AccountNotFoundException(AccountConstant.NOT_FOUND));
 
-        return accountMapper.accountToAccountDto(account);
-    }
+    return accountMapper.accountToAccountDto(account);
+  }
 
-    @Override
-    public AccountDto verifyAccount(String token) {
-        Account account = verificationTokenService.verifyToken(token);
-        account.setVerified(true);
-        Account updatedAccount = accountRepository.save(account);
-
-        return accountMapper.accountToAccountDto(updatedAccount);
-    }
+  @Override
+  public AccountDto verifyAccount(String token) {
+    Account account = verificationTokenService.verifyToken(token);
+    account.setVerified(true);
+    Account updatedAccount = accountRepository.save(account);
 
-    @Override
-    public LoginResponse login(LoginRequest loginRequest) {
-        authenticationManager.authenticate(
-                new UsernamePasswordAuthenticationToken(
-                        loginRequest.getEmail(),
-                        loginRequest.getPassword()
-                )
-        );
+    return accountMapper.accountToAccountDto(updatedAccount);
+  }
 
-        Account account = accountRepository.findOneByEmail(loginRequest.getEmail())
-                .orElseThrow(() -> new AccountNotFoundException(AccountConstant.NOT_FOUND));
+  @Override
+  public LoginResponse login(LoginRequest loginRequest) {
+    authenticationManager.authenticate(
+        new UsernamePasswordAuthenticationToken(
+            loginRequest.getEmail(),
+            loginRequest.getPassword()
+        )
+    );
 
-        if (!account.isVerified()) {
-            throw new AccountNotVerifiedException(AccountConstant.NOT_VERIFIED);
-        }
+    Account account = findAccountByEmail(loginRequest.getEmail());
 
-        return new LoginResponse(jwtService.generateToken(account));
+    if (!account.isVerified()) {
+      throw new AccountNotVerifiedException(AccountConstant.NOT_VERIFIED);
     }
 
-    private void sendVerificationMail(String receiver, String token) {
-        String link = baseUrl + verifyUrl + token;
-        byte[] contentBytes = fileService.readHtml(FilePathConstant.VERIFICATION_EMAIL_HTML);
-        String content = new String(contentBytes).replace(HtmlConstant.HREF, link);
+    return new LoginResponse(jwtService.generateToken(account));
+  }
 
-        emailService.send(receiver, content);
+  @Override
+  public AccountDto updateAccount(AccountDto accountDto, Principal principal) {
+    Account account = findAccountByEmail(accountDto.getEmail());
```

Comment:
Here we can easily get null pointer exception, because you didn't validate the supplied accountDto. In the controller above.